SupportYourLocals.Data/CSVData.cs

```diff
@@ -5,73 +5,95 @@
+                return new List<UserData>();                
+            }
+            else
+            {
+                var userDataList = new List<UserData>();
+                using (TextFieldParser csvParser = new TextFieldParser(filePath))
+                {
+                    csvParser.CommentTokens = new[] { "#" };
+                    csvParser.SetDelimiters(new[] { "," });
+                    csvParser.HasFieldsEnclosedInQuotes = true;
 
-            personsID = GetPersonsID(personsID);
+                    // Skip the row with the column names
+                    csvParser.ReadLine();
 
-            var newLine = "{0},{1},{2}".Format(product, position, personsID);
+                    while (!csvParser.EndOfData)
+                    {
+                        string[] fields = csvParser.ReadFields();
 
-            csv.AppendLine(newLine);
+                        var username = fields[0]; 
+                        var passwordHash = fields[1]; 
+                        var salt = fields[2]; 
+                        var id = int.Parse(fields[3]);
 
-            File.AppendAllText(filePath, csv.ToString());
+                        userDataList.Add(new UserData(username, passwordHash, salt, id));
+                    }
+                }
+                return userDataList;
+            }            
         }
 
-        public static int GetPersonsID(int personsID)
+        public void SaveData()
         {
-            // Checking person's ID
-            List<double> personID = new List<double>();
-            using (TextFieldParser csvParser = new TextFieldParser(filePath))
-            {
-                csvParser.CommentTokens = new[] { "#" };
-                csvParser.SetDelimiters(new[] { "," });
-                csvParser.HasFieldsEnclosedInQuotes = true;
+            StringBuilder csv = new StringBuilder();
 
-                // Skip the row with the column names
-                csvParser.ReadLine();
+            if (!File.Exists(filePath))
+            {
+                csv.AppendLine("Username, Hashed password, Salt, ID");
+            }
 
-                while (!csvParser.EndOfData)
-                {
-                    // Read current line fields, pointer moves to the next line.
-                    string[] fields = csvParser.ReadFields();
-                    personID.Add(double.Parse(fields[3]));
-                }
+            foreach (var user in listUserData)
+            {
+                var newLine = "{0},{1},{2},{3}".Format(user.Username, user.PasswordHash, user.Salt, user.ID);
+                csv.AppendLine(newLine);
             }
 
-            // Setting person's ID
-            personsID += personID.Count;
-            return personsID;
-        }
+            File.AppendAllText(filePath, csv.ToString());
+        }        
 
-        public static void SetMarkers(List<double> listXCoord, List<double> listYCoord, List<int> listPersonsID)
+        public UserData GetData(string username)
```

Comment:
This function could be a single line if a dictionary was used instead of a list. (Also should be id, not username)
public UserData GetData(string id) => dictionaryUserData[id];